### ISSUE TITLE
reference installed headers from sysapi / tcti

### ISSUE THREAD

**Issue by flihp** [01org/tpm2.0-tools#13]
In reviewing [PR #92](https://github.com/01org/TPM2.0-TSS/pull/92) from @theopolis in the TPM2.0-TSS repo I was surprised that this issue didn't surface when we starting linking against libtpm2sapi / libtpm2tcti. Turns out the tpm2.0-tools build pulls headers directly from the TPM2.0-TSS tree that it checks out by putting them on the search path for gcc. So basically we're building against headers and linking against libraries from different locations. This is a disaster waiting to happen :(

I don't know that there's an easy fix here. Every one of the TPM tools builds and links against code from the 'common' and 'test' directories from the TSS repo. This means simply changing the gcc search path won't solve the problem. This will likely require coordinated work across the repos including some cleanup of my past mistakes in the TPM2.0-TSS build / include approach.

More issues to come in this cleanup in both repos.

**Comment by theopolis** [comment]
Hey @flihp, somewhat related, is it possible to get some CI going for the https://github.com/01org/TPM2.0-TSS? I'd imagine TravisCI's Ubuntu 12.04 could build, `make install`, launch the simulator and run the suite of tests. For Windows https://www.appveyor.com/ is the recommended option, to do the same build/test workflow.

**Comment by flihp** [comment]
I've brought this up in the past but no one has taken the lead, myself included. I had a buildbot instance building the TPM2.0-TSS for a while. It went down for maintenance for one of my other projects and never went back up sadly.

While we're building an infrastructure wishlist, do you think there would be any use for a mailing list or maybe a google group to keep maintenance down?

**Comment by theopolis** [comment]
+1 for a mailing list, or channel on Freenode.

In the mean time wrt build, I can set up a Travis/Appveyor for my fork, if all goes well and you decide it works for the project I don't mind being oncall for fixes. In a more perfect world a builtbot or some incarnation of Jenkins could run more-integration-like tests on machines will fTPMs.

**Comment by flihp** [comment]
My brain went to IRC channel immediately after my last comment. See #tpm2.0-tss on freenode. I'm camping there currently.

**Comment by gwei3** [comment]
I joined  #tpm2.0-tss on freenode too, but not sure to be able to keep active status with your at the same period.

**Comment by flihp** [comment]
The discussion that will likely resolve this issue is over in #5. I spent some time this week trying to come up with a way to fix this without doing anything too drastic (either copying the common code over to the tools repo, or moving the tools code over to the tss repo). Everything I came up with caused more problems than it solved and was generally horrible. I think it's best to close down this issue and keep the discussion in #5.
